{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[{"file_id":"1B0iROHGEOjXPwEQpFviWANYA5LQfEn6g","timestamp":1711670622366}]},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"markdown","source":["#### Here is the central spot for the projects datasets"],"metadata":{"id":"qC2c-vtT97i6"}},{"cell_type":"code","source":["###### CRISPR sample\n","\n","!gdown 1-96T1PZKA_FQeD_ZK5z9USaHLP3jVdRO\n","!gdown 1-BXGr3XVGtd9Tx6PCHSp4hepK41GGCTS\n","\n","### To make thinngs faster for the student, here is the indexed mouse genome, mm10.\n","### You have to donwload it and decompressed it, you will get both the fasta and the BWT'ed files. Note all have same prefix Mouse<.files>.\n","### Next semester we will cover how specialized computers are often used to solve this aligmnet problems faster.\n","\n","!gdown 1a8CP4P5zkzIBiw1EleqJiSwDW0VZcAar"],"metadata":{"id":"W_X1Axpa_oJ7","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1713403411649,"user_tz":420,"elapsed":46120,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}},"outputId":"ee61bd3b-f26b-42b9-ad52-6f781e5cbcd0"},"execution_count":10,"outputs":[{"output_type":"stream","name":"stdout","text":["Downloading...\n","From (original): https://drive.google.com/uc?id=1-96T1PZKA_FQeD_ZK5z9USaHLP3jVdRO\n","From (redirected): https://drive.google.com/uc?id=1-96T1PZKA_FQeD_ZK5z9USaHLP3jVdRO&confirm=t&uuid=22993a89-ee49-4ebe-9f34-b2bfb9e9412c\n","To: /content/CRISPR.R1.fastq\n","  8% 202M/2.67G [00:02<00:38, 64.1MB/s]Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/urllib3/response.py\", line 710, in _error_catcher\n","    yield\n","  File \"/usr/local/lib/python3.10/dist-packages/urllib3/response.py\", line 814, in _raw_read\n","    data = self._fp_read(amt) if not fp_closed else b\"\"\n","  File \"/usr/local/lib/python3.10/dist-packages/urllib3/response.py\", line 799, in _fp_read\n","    return self._fp.read(amt) if amt is not None else self._fp.read()\n","  File \"/usr/lib/python3.10/http/client.py\", line 466, in read\n","    s = self.fp.read(amt)\n","  File \"/usr/lib/python3.10/socket.py\", line 705, in readinto\n","    return self._sock.recv_into(b)\n","  File \"/usr/lib/python3.10/ssl.py\", line 1303, in recv_into\n","    return self.read(nbytes, buffer)\n","  File \"/usr/lib/python3.10/ssl.py\", line 1159, in read\n","    return self._sslobj.read(len, buffer)\n","KeyboardInterrupt\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/bin/gdown\", line 8, in <module>\n","    sys.exit(main())\n","  File \"/usr/local/lib/python3.10/dist-packages/gdown/cli.py\", line 169, in main\n","    download(\n","  File \"/usr/local/lib/python3.10/dist-packages/gdown/download.py\", line 325, in download\n","    for chunk in res.iter_content(chunk_size=CHUNK_SIZE):\n","  File \"/usr/local/lib/python3.10/dist-packages/requests/models.py\", line 816, in generate\n","    yield from self.raw.stream(chunk_size, decode_content=True)\n","  File \"/usr/local/lib/python3.10/dist-packages/urllib3/response.py\", line 936, in stream\n","    data = self.read(amt=amt, decode_content=decode_content)\n","  File \"/usr/local/lib/python3.10/dist-packages/urllib3/response.py\", line 879, in read\n","    data = self._raw_read(amt)\n","  File \"/usr/local/lib/python3.10/dist-packages/urllib3/response.py\", line 813, in _raw_read\n","    with self._error_catcher():\n","  File \"/usr/lib/python3.10/contextlib.py\", line 153, in __exit__\n","    self.gen.throw(typ, value, traceback)\n","  File \"/usr/local/lib/python3.10/dist-packages/urllib3/response.py\", line 740, in _error_catcher\n","    self._original_response.close()\n","  File \"/usr/lib/python3.10/http/client.py\", line 422, in close\n","    self._close_conn()\n","KeyboardInterrupt\n","  8% 210M/2.67G [00:02<00:33, 73.9MB/s]\n","^C\n","Downloading...\n","From (original): https://drive.google.com/uc?id=1-BXGr3XVGtd9Tx6PCHSp4hepK41GGCTS\n","From (redirected): https://drive.google.com/uc?id=1-BXGr3XVGtd9Tx6PCHSp4hepK41GGCTS&confirm=t&uuid=cf789027-ef56-4262-814a-a2033f5c7a3b\n","To: /content/CRISPR.R2.fastq\n"," 14% 363M/2.67G [00:02<00:15, 149MB/s]Traceback (most recent call last):\n","  File \"/usr/local/bin/gdown\", line 8, in <module>\n","    sys.exit(main())\n","  File \"/usr/local/lib/python3.10/dist-packages/gdown/cli.py\", line 169, in main\n","    download(\n","  File \"/usr/local/lib/python3.10/dist-packages/gdown/download.py\", line 326, in download\n","    f.write(chunk)\n","KeyboardInterrupt\n"," 14% 375M/2.67G [00:02<00:16, 141MB/s]\n","^C\n","Downloading...\n","From (original): https://drive.google.com/uc?id=1a8CP4P5zkzIBiw1EleqJiSwDW0VZcAar\n","From (redirected): https://drive.google.com/uc?id=1a8CP4P5zkzIBiw1EleqJiSwDW0VZcAar&confirm=t&uuid=0004582e-a015-44b9-8a86-00eb65fd05c7\n","To: /content/MM10.tar.gz\n","100% 3.91G/3.91G [00:37<00:00, 105MB/s]\n"]}]},{"cell_type":"code","source":["### 2 fastq files\n","### 1 Reference genome\n","### fastq files -> BAM/SAM -> vcf -> df -> chr2 -> 9 position ->\n","                            ## -> off-target site\n","!apt install bwa\n","!apt install samtools\n","!apt install bcftools"],"metadata":{"id":"i3M7KNUFTwMo","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1713404440614,"user_tz":420,"elapsed":12344,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}},"outputId":"85e20961-69ee-4351-9300-373198473be5"},"execution_count":30,"outputs":[{"output_type":"stream","name":"stdout","text":["Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","bwa is already the newest version (0.7.17-6).\n","0 upgraded, 0 newly installed, 0 to remove and 45 not upgraded.\n","Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","samtools is already the newest version (1.13-4).\n","0 upgraded, 0 newly installed, 0 to remove and 45 not upgraded.\n","Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","bcftools is already the newest version (1.13-1).\n","0 upgraded, 0 newly installed, 0 to remove and 45 not upgraded.\n"]}]},{"cell_type":"code","source":["ls"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"gFkAbpXx0E0Q","executionInfo":{"status":"ok","timestamp":1713404593980,"user_tz":420,"elapsed":210,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}},"outputId":"30b336b3-eeaf-4f0f-e7c3-3517f9313d20"},"execution_count":35,"outputs":[{"output_type":"stream","name":"stdout","text":["CRISPR.bam  CRISPR.R1.sai  CRISPR.R2.sai  \u001b[0m\u001b[01;34mdrive\u001b[0m/  \u001b[01;34mMM10\u001b[0m/  MM10.tar.gz  \u001b[01;34msample_data\u001b[0m/\n"]}]},{"cell_type":"code","source":["!tar -xf MM10.tar.gz"],"metadata":{"id":"wbsDDQv8Z2sy","executionInfo":{"status":"ok","timestamp":1713403852030,"user_tz":420,"elapsed":150907,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}}},"execution_count":24,"outputs":[]},{"cell_type":"code","source":["ls MM10/Mouse"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"7qzrLOx91jW1","executionInfo":{"status":"ok","timestamp":1713404590324,"user_tz":420,"elapsed":154,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}},"outputId":"18faa62b-0086-42f8-c04e-2b53f84a6e57"},"execution_count":34,"outputs":[{"output_type":"stream","name":"stdout","text":["CRISPR.R1.fastq  CRISPR.R2.fastq\n"]}]},{"cell_type":"code","source":["!bwa aln MM10/Mouse CRISPR.R1.fastq > CRISPR.R1.sai\n","!bwa aln MM10/Mouse CRISPR.R2.fastq > CRISPR.R2.sai\n","!bwa sampe MM10/Mouse CRISPR.R1.sai CRISPR.R2.sai  CRISPR.R1.fastq CRISPR.R2.fastq > CRISPR.bam"],"metadata":{"id":"pcToRCRPfYmA","outputId":"667e7b18-97b2-47b4-d7e7-1d74fa3d47da","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1713404587590,"user_tz":420,"elapsed":527,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}}},"execution_count":33,"outputs":[{"output_type":"stream","name":"stdout","text":["[bwa_aln] 17bp reads: max_diff = 2\n","[bwa_aln] 38bp reads: max_diff = 3\n","[bwa_aln] 64bp reads: max_diff = 4\n","[bwa_aln] 93bp reads: max_diff = 5\n","[bwa_aln] 124bp reads: max_diff = 6\n","[bwa_aln] 157bp reads: max_diff = 7\n","[bwa_aln] 190bp reads: max_diff = 8\n","[bwa_aln] 225bp reads: max_diff = 9\n","[bwa_seq_open] fail to open file 'CRISPR.R1.fastq' : No such file or directory\n","[bwa_aln] 17bp reads: max_diff = 2\n","[bwa_aln] 38bp reads: max_diff = 3\n","[bwa_aln] 64bp reads: max_diff = 4\n","[bwa_aln] 93bp reads: max_diff = 5\n","[bwa_aln] 124bp reads: max_diff = 6\n","[bwa_aln] 157bp reads: max_diff = 7\n","[bwa_aln] 190bp reads: max_diff = 8\n","[bwa_aln] 225bp reads: max_diff = 9\n","[bwa_seq_open] fail to open file 'CRISPR.R2.fastq' : No such file or directory\n","[fread] Unexpected end of file\n"]}]},{"cell_type":"code","source":["!bwa aln CRISPR.R1.fastq > CRISPR.R1.sai\n","!bwa aln CRISPR.R2.fastq > CRISPR.R2.sai\n","!bwa sampe CRISPR.R1.sai CRISPR.R2.sai CRISPR.R1.fastq CRISPR.R2.fastq > CRISPR.bam"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"gi2aTPaO4Dwo","executionInfo":{"status":"ok","timestamp":1713404664082,"user_tz":420,"elapsed":527,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}},"outputId":"34b54132-006c-486d-923e-cdeaed9b3f20"},"execution_count":36,"outputs":[{"output_type":"stream","name":"stdout","text":["\n","Usage:   bwa aln [options] <prefix> <in.fq>\n","\n","Options: -n NUM    max #diff (int) or missing prob under 0.02 err rate (float) [0.04]\n","         -o INT    maximum number or fraction of gap opens [1]\n","         -e INT    maximum number of gap extensions, -1 for disabling long gaps [-1]\n","         -i INT    do not put an indel within INT bp towards the ends [5]\n","         -d INT    maximum occurrences for extending a long deletion [10]\n","         -l INT    seed length [32]\n","         -k INT    maximum differences in the seed [2]\n","         -m INT    maximum entries in the queue [2000000]\n","         -t INT    number of threads [1]\n","         -M INT    mismatch penalty [3]\n","         -O INT    gap open penalty [11]\n","         -E INT    gap extension penalty [4]\n","         -R INT    stop searching when there are >INT equally best hits [30]\n","         -q INT    quality threshold for read trimming down to 35bp [0]\n","         -f FILE   file to write output to instead of stdout\n","         -B INT    length of barcode\n","         -L        log-scaled gap penalty for long deletions\n","         -N        non-iterative mode: search for all n-difference hits (slooow)\n","         -I        the input is in the Illumina 1.3+ FASTQ-like format\n","         -b        the input read file is in the BAM format\n","         -0        use single-end reads only (effective with -b)\n","         -1        use the 1st read in a pair (effective with -b)\n","         -2        use the 2nd read in a pair (effective with -b)\n","         -Y        filter Casava-filtered sequences\n","\n","\n","Usage:   bwa aln [options] <prefix> <in.fq>\n","\n","Options: -n NUM    max #diff (int) or missing prob under 0.02 err rate (float) [0.04]\n","         -o INT    maximum number or fraction of gap opens [1]\n","         -e INT    maximum number of gap extensions, -1 for disabling long gaps [-1]\n","         -i INT    do not put an indel within INT bp towards the ends [5]\n","         -d INT    maximum occurrences for extending a long deletion [10]\n","         -l INT    seed length [32]\n","         -k INT    maximum differences in the seed [2]\n","         -m INT    maximum entries in the queue [2000000]\n","         -t INT    number of threads [1]\n","         -M INT    mismatch penalty [3]\n","         -O INT    gap open penalty [11]\n","         -E INT    gap extension penalty [4]\n","         -R INT    stop searching when there are >INT equally best hits [30]\n","         -q INT    quality threshold for read trimming down to 35bp [0]\n","         -f FILE   file to write output to instead of stdout\n","         -B INT    length of barcode\n","         -L        log-scaled gap penalty for long deletions\n","         -N        non-iterative mode: search for all n-difference hits (slooow)\n","         -I        the input is in the Illumina 1.3+ FASTQ-like format\n","         -b        the input read file is in the BAM format\n","         -0        use single-end reads only (effective with -b)\n","         -1        use the 1st read in a pair (effective with -b)\n","         -2        use the 2nd read in a pair (effective with -b)\n","         -Y        filter Casava-filtered sequences\n","\n","\n","Usage:   bwa sampe [options] <prefix> <in1.sai> <in2.sai> <in1.fq> <in2.fq>\n","\n","Options: -a INT   maximum insert size [500]\n","         -o INT   maximum occurrences for one end [100000]\n","         -n INT   maximum hits to output for paired reads [3]\n","         -N INT   maximum hits to output for discordant pairs [10]\n","         -c FLOAT prior of chimeric rate (lower bound) [1.0e-05]\n","         -f FILE  sam file to output results to [stdout]\n","         -r STR   read group header line such as `@RG\\tID:foo\\tSM:bar' [null]\n","         -P       preload index into memory (for base-space reads only)\n","         -s       disable Smith-Waterman for the unmapped mate\n","         -A       disable insert size estimate (force -s)\n","\n","Notes: 1. For SOLiD reads, <in1.fq> corresponds R3 reads and <in2.fq> to F3.\n","       2. For reads shorter than 30bp, applying a smaller -o is recommended to\n","          to get a sensible speed at the cost of pairing accuracy.\n","\n"]}]},{"cell_type":"code","source":[],"metadata":{"id":"cjW0rQTrfbPD"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["from google.colab import drive\n","drive.mount('/content/drive')"],"metadata":{"id":"AGpOQv0genUp","executionInfo":{"status":"ok","timestamp":1713403249572,"user_tz":420,"elapsed":18027,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}},"outputId":"333cc157-2ccc-445f-f7a9-869155fb9144","colab":{"base_uri":"https://localhost:8080/"}},"execution_count":1,"outputs":[{"output_type":"stream","name":"stdout","text":["Mounted at /content/drive\n"]}]},{"cell_type":"code","source":["!bwa index -p ###.fasta文件"],"metadata":{"id":"t1gklg89W033","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1713403427527,"user_tz":420,"elapsed":153,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}},"outputId":"5c2d52e3-3ff8-4f80-b388-b0ebe09c298d"},"execution_count":13,"outputs":[{"output_type":"stream","name":"stdout","text":["index: option requires an argument -- 'p'\n"]}]},{"cell_type":"code","source":["!head -n 100 CRISPR.bam #检验 ##.bam是否为空档"],"metadata":{"id":"E_4IsEOFaY0O","executionInfo":{"status":"ok","timestamp":1713403430679,"user_tz":420,"elapsed":179,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}}},"execution_count":14,"outputs":[]},{"cell_type":"code","source":["!samtools view -bS -q 12 CRISPR.bam | samtools sort -o CRISPR.bam.sorted"],"metadata":{"id":"oG6vFm5ya2Fj","executionInfo":{"status":"ok","timestamp":1713403436884,"user_tz":420,"elapsed":145,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}},"outputId":"c350a0c7-81fd-4824-f05a-4a884ac45ac6","colab":{"base_uri":"https://localhost:8080/"}},"execution_count":15,"outputs":[{"output_type":"stream","name":"stdout","text":["[main_samview] fail to read the header from \"CRISPR.bam\".\n","samtools sort: failed to read header from \"-\"\n"]}]},{"cell_type":"code","source":["!samtools view CRISPR.bam.sorted"],"metadata":{"id":"XTDRCgXnbVmo","executionInfo":{"status":"ok","timestamp":1713403439266,"user_tz":420,"elapsed":462,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}},"outputId":"02f19bb9-adea-4b8e-b9a9-3c40ac2314b8","colab":{"base_uri":"https://localhost:8080/"}},"execution_count":16,"outputs":[{"output_type":"stream","name":"stdout","text":["[E::hts_open_format] Failed to open file \"CRISPR.bam.sorted\" : No such file or directory\n","samtools view: failed to open \"CRISPR.bam.sorted\" for reading: No such file or directory\n"]}]},{"cell_type":"code","source":["!bcftools mpileup -f Mouse.fasta --max-depth  2000 CRISPR.bam.sorted  | bcftools call --multiallelic-caller --variants-only --ploidy 1  -mv -Oz -o Variants.vcf.gz"],"metadata":{"id":"8I_YCCFrdtsg","executionInfo":{"status":"ok","timestamp":1713403440885,"user_tz":420,"elapsed":207,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}},"outputId":"3cc31b7b-da0b-4cef-84fb-6c75c604ef9e","colab":{"base_uri":"https://localhost:8080/"}},"execution_count":17,"outputs":[{"output_type":"stream","name":"stdout","text":["[E::fai_build3_core] Failed to open the file Mouse.fasta\n","Failed to read from standard input: unknown file type\n"]}]},{"cell_type":"code","source":["!gzip -df Variants.vcf.gz"],"metadata":{"id":"JQAri_ctj8Bl","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1713403443183,"user_tz":420,"elapsed":163,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}},"outputId":"b11f45ad-f3b4-40dc-a71a-2627577a6483"},"execution_count":18,"outputs":[{"output_type":"stream","name":"stdout","text":["gzip: Variants.vcf.gz: No such file or directory\n"]}]},{"cell_type":"code","source":["!cat Variants.vcf"],"metadata":{"id":"4czWLz_fkBMl","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1713403446654,"user_tz":420,"elapsed":242,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}},"outputId":"6cff6ad3-f40e-4ab4-ff1d-f36b59836d70"},"execution_count":19,"outputs":[{"output_type":"stream","name":"stdout","text":["cat: Variants.vcf: No such file or directory\n"]}]},{"cell_type":"code","source":["!pip install cyvcf2\n","#!gzip -df SARS.Variants.vcf.gz\n","vcf_path = 'Variants.vcf'"],"metadata":{"id":"RB5zvd4WkGtl","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1713403461546,"user_tz":420,"elapsed":13437,"user":{"displayName":"Xintian Lyu","userId":"14871865838627291827"}},"outputId":"cd535ff3-e4ef-4923-86d6-a78070faec38"},"execution_count":20,"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting cyvcf2\n","  Downloading cyvcf2-0.30.28-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (6.8 MB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m6.8/6.8 MB\u001b[0m \u001b[31m12.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from cyvcf2) (1.25.2)\n","Collecting coloredlogs (from cyvcf2)\n","  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m46.0/46.0 kB\u001b[0m \u001b[31m3.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: click in /usr/local/lib/python3.10/dist-packages (from cyvcf2) (8.1.7)\n","Collecting humanfriendly>=9.1 (from coloredlogs->cyvcf2)\n","  Downloading humanfriendly-10.0-py2.py3-none-any.whl (86 kB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m86.8/86.8 kB\u001b[0m \u001b[31m7.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hInstalling collected packages: humanfriendly, coloredlogs, cyvcf2\n","Successfully installed coloredlogs-15.0.1 cyvcf2-0.30.28 humanfriendly-10.0\n"]}]},{"cell_type":"code","source":["import cyvcf2\n","\n","# Create a VCF reader object\n","vcf_reader = cyvcf2.VCF(vcf_path)\n","\n","with open('output.tsv', 'w') as file:\n","# Loop through each variant in the VCF file\n","    for variant in vcf_reader:\n","    # Access various properties of the variant\n","    ### Se note below\n","        genotype = variant.gt_types[0]\n","        data_string = f\"{variant.CHROM}\\t{variant.POS}\\t{variant.ID}\\t{variant.REF}\\t{','.join(variant.ALT)}\\t{variant.QUAL}\\t{variant.FILTER}\\t{genotype}\\n\"\n","        # Write the data string to the file\n","        file.write(data_string)"],"metadata":{"id":"7r_ZX82skR2u"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["import pandas"],"metadata":{"id":"kE4AV54pmakd"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["columns = ['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'GENOTYPE']\n","df = pandas.read_csv(\"output.tsv\", delimiter = \"\\t\", header=None, names=columns)\n","\n","df"],"metadata":{"id":"zxH7Fd9kme-N"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["df_chr2 = df[df['CHROM'] == 'chr2']\n","df_chr2"],"metadata":{"id":"Lp4YKRU1mRkG"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["pos = ['36937210',\n","'36996899',\n","'85400441',\t'85776687',\n","'85918029',\n","'86198668',\t'86236802',\n","'86658391',\n","'87049235']\n","df_target = df_chr2[df_chr2['POS'].isin(pos)]\n","df_target"],"metadata":{"id":"EpRCRbT9mzOf"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":[],"metadata":{"id":"a2QRKDVTXFxW"}},{"cell_type":"markdown","source":[],"metadata":{"id":"USi81_YDBvVm"}}]}